### PR TITLE
DO-188: Completed functionality for add address in edit profile.

### PR DESCRIPTION
**Pull request by ayushmalik2110**

### CODE REVIEW

File: src/components/member/MemberDetailsWizard/CustomForm/AddressForm/addressFormSchema.json [hunk 1]
```diff
@@ -0,0 +1,169 @@
+{
```
Comment (mohitsaladiya) [inline src/components/member/MemberDetailsWizard/CustomForm/AddressForm/addressFormSchema.json:1]: Mention this json under confluence in the Ui guidelines resources table.

File: src/utils/member/utils.js [hunk 2]
```diff
@@ -23,8 +23,8 @@ export const getMemberImageDefaultUrl = (member = {}) => {
  * @param {Array} dpImageUrlList
  * @return {Array}
  */
-export const updateImageUrlForMemberList = ({ members = [], dpImageUrlList = [] }) => {
-  return members.map(member => {
+export const updateImageUrlForMemberList
```
Comment (mohitsaladiya) [inline src/utils/member/utils.js:26]: Please remove this method. I think we're getting defaultImage urls from back-end

File: src/utils/member/utils.js [hunk 3]
```diff
@@ -48,3 +48,385 @@ export const updateImageUrlForMemberList = ({ members = [], dpImageUrlList = []
...
+  addressOneSchema,
+}) => {
+  // Creating a copy of form schema.
+  let updatedFormSchema = formSchema;
+
+  // Setting the show add adress button to true.
+  let showAddAddressButton = true;
+
+  // Checking if schema has addressOne node
+  if (!formSchema.properties.address.properties.addressOne) {
```
Comment (mohitsaladiya) [inline src/utils/member/utils.js:100]: Use destructuring

File: src/utils/member/utils.js [hunk 4]
```diff
@@ -48,3 +48,385 @@ export const updateImageUrlForMemberList = ({ members = [], dpImageUrlList = []
...
+            ...updatedFormSchema.properties.address.properties,
+            addressOne: addressOneSchema,
+          },
+        },
+
+      },
+    };
+    // If addressOne is already added and if use click on add button again.
+    // Then we are checking for addressTwo node.
+  } else if (!formSchema.properties.address.properties.addressTwo) {
```
Comment (mohitsaladiya) [inline src/utils/member/utils.js:118]: Here as well

File: src/utils/member/utils.js [hunk 5]
```diff
@@ -48,3 +48,385 @@ export const updateImageUrlForMemberList = ({ members = [], dpImageUrlList = []
...
+
+  return updatedFormData;
+};
+
+/**
+ * This util function format the address detail for edit form in profile section.
+ * @param {Object} formData
+ * @return {{address: {addressTwo: *, addressOne: *}}|{address: {addressTwo: {}, addressOne: {}}}}
+ */
+export const formatAddressFormDataForEdit = (formData = {}) => {
```
Comment (mohitsaladiya) [inline src/utils/member/utils.js:337]:
```suggestion
export const formatAddressFormDataForEditProfile = (formData = {}) => {

```

OR

```suggestion
export const formatAddressFormData = (formData = {}) => {

```

File: src/utils/member/utils.js [hunk 6]
```diff
@@ -48,3 +48,385 @@ export const updateImageUrlForMemberList = ({ members = [], dpImageUrlList = []
...
+
+/**
+ * This util function mark an object as empty if it's all field is undefined
+ * or null. You also skip values that you don't want to check, just simply
+ * pass that value key into the exclude field array.
+ * @param {Object} data
+ * @param {Array} excludeField
+ * @return {boolean}
+ */
+export const isObjectEmpty = (data = {}, excludeField = []) => {
```
Comment (mohitsaladiya) [inline src/utils/member/utils.js:60]: We can rename it to something like
```suggestion
export const isObjectRequiredKeysUnavailable = (data = {}, excludeField = []) => {

```

File: src/components/member/MemberDetailsWizard/CustomForm/AddressForm/AddressForm.jsx [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+      }
+    }
+
+    // This code fragment is to show hide address two when user isAddressTwoSame
+    // set to true/false
+    if (!isEmpty(formData.address)) {
+      const isAddressTwoSameCurrent = formData.address.addressOne.isAddressTwoSame;
+      const { addressOne: { isAddressTwoSame: isAddressTwoSamePrevious } = {} }
+      = StateFormData.address;
+      if (isAddressTwoSameCurrent !== isAddressTwoSamePrevious) {
```
Comment (mohitsaladiya) [inline src/components/member/MemberDetailsWizard/CustomForm/AddressForm/AddressForm.jsx:156]: Rename it to
```suggestion
      if (isAddressTwoSameAsCurrentAddress !== isAddressTwoSameAsPreviousAddress) {
```

File: src/components/member/MemberDetailsWizard/CustomForm/AddressForm/AddressForm.jsx [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+    const { formSchema, formData: StateFormData, showAddAddressButton } = this.state;
+    let showAddAddressButtonState = showAddAddressButton;
+    let updatedFormSchema = formSchema;
+    let updatedFormData = formData;
+
+    // This code fragment is to change address two drop down when
+    // address type for address one change.
+    if (!isEmpty(formSchema.properties.address.properties.addressOne)
+      && formSchema.properties.address.properties.addressTwo) {
+      const addressOneTypeCurrent = formData.address.addressOne.addressType;
```
Comment (mohitsaladiya) [inline src/components/member/MemberDetailsWizard/CustomForm/AddressForm/AddressForm.jsx:102]: Simply call it
```suggestion
      const addressOneType = formData.address.addressOne.addressType;
```

File: src/components/member/MemberDetailsWizard/CustomForm/AddressForm/AddressForm.jsx [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+  // then remove the second address node.
+  onChangeFormData = ({ formData }) => {
+    const { formSchema, formData: StateFormData, showAddAddressButton } = this.state;
+    let showAddAddressButtonState = showAddAddressButton;
+    let updatedFormSchema = formSchema;
+    let updatedFormData = formData;
+
+    // This code fragment is to change address two drop down when
+    // address type for address one change.
+    if (!isEmpty(formSchema.properties.address.properties.addressOne)
```
[the remark this answers is not in the document]
Comment (ayushmalik2110) [reply src/components/member/MemberDetailsWizard/CustomForm/AddressForm/AddressForm.jsx:100]: Done

File: src/components/member/MemberDetailsWizard/CustomForm/AddressForm/AddressForm.jsx (hunk 7 above)
Comment (ayushmalik2110) [reply src/components/member/MemberDetailsWizard/CustomForm/AddressForm/AddressForm.jsx:156]: Done

File: src/components/member/MemberDetailsWizard/CustomForm/AddressForm/AddressForm.jsx (hunk 8 above)
Comment (ayushmalik2110) [reply src/components/member/MemberDetailsWizard/CustomForm/AddressForm/AddressForm.jsx:102]: done

File: src/components/member/MemberDetailsWizard/CustomForm/AddressForm/addressFormSchema.json (hunk 1 above)
Comment (ayushmalik2110) [reply src/components/member/MemberDetailsWizard/CustomForm/AddressForm/addressFormSchema.json:1]: Done

File: src/utils/member/utils.js (hunk 2 above)
Comment (ayushmalik2110) [reply src/utils/member/utils.js:27]: done.

File: src/utils/member/utils.js (hunk 3 above)
Comment (ayushmalik2110) [reply src/utils/member/utils.js:101]: done

File: src/utils/member/utils.js (hunk 4 above)
Comment (ayushmalik2110) [reply src/utils/member/utils.js:119]: done

File: src/utils/member/utils.js (hunk 5 above)
Comment (ayushmalik2110) [reply src/utils/member/utils.js:340]: done

File: src/utils/member/utils.js (hunk 6 above)
Comment (ayushmalik2110) [reply src/utils/member/utils.js]: done

**Pull request merged**